controllers/metrics.go

```diff
@@ -0,0 +1,196 @@
+package controllers
+
+import (
+	"context"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/common/log"
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+var (
+	// 'backupPhase' - [New, FailedValidation, InProgress, Uploading, UploadingPartialFailure, Completed, PartiallyFailed, Failed, Deleting]
+	// 'restorePhase' - [New, FailedValidation, InProgress, Completed, PartiallyFailed, Failed]
+	backupGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "oadp_app_workload_backup_total",
+		Help: "Count of OADP backups sorted by phase",
+	},
+		[]string{"phase"},
+	)
+
+	restoreGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "oadp_app_workload_restore_total",
+		Help: "Count of OADP restores sorted by phase",
+	},
+		[]string{"phase"},
+	)
+)
+
+func recordBackupMetrics(client client.Client) {
```

Comment:
it looks like you're writing an exporter around the velerov1 resources. Why can't the operand implement the metrics directly? If it's not possible, you probably need to use constant metrics as described in the [client-go](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#hdr-Custom_Collectors_and_constant_Metrics) doc.

Reply:
> On finding this out we decided that we would like to capture these metrics as well as introduce a new gauge metric for our operator


I'm not sure to understand why you want the metrics to be exposed by both Velero and the OADP operator.

As I hinted in the first comment, I would recommend to use a [custom collector](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#hdr-Custom_Collectors_and_constant_Metrics) instead of a goroutine polling the Kube API and updating the metrics out-of-band.